src/pytest_bdd/gherkin_parser.py

```diff
@@ -1,56 +1,94 @@
 import linecache
 import textwrap
-from pathlib import Path
-from typing import List, Optional
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
 
 from gherkin.errors import CompositeParserException
 from gherkin.parser import Parser
 from gherkin.token_scanner import TokenScanner
-from pydantic import BaseModel, field_validator, model_validator
 
 from . import exceptions
 from .types import STEP_TYPES
 
 
-class Location(BaseModel):
+@dataclass
+class Location:
     column: int
     line: int
 
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "Location":
```

Comment:
let's use `from __future__ import annotations`, so that here we can do:

```suggestion
    def from_dict(cls, data: Dict[str, Any]) -> Self:
```